src/System.Management.Automation/engine/parser/CharTraits.cs

```diff
@@ -376,11 +376,16 @@ internal static bool ForceStartNewToken(this char c)
 
         // Return true if the character ends the current number token.  This allows the tokenizer
         // to scan '7z' as a single token, but '7+' as 2 tokens.
-        internal static bool ForceStartNewTokenAfterNumber(this char c)
+        internal static bool ForceStartNewTokenAfterNumber(this char c, bool forceEndNumberOnTernaryOperatorChars)
         {
             if (c < 128)
             {
-                return (s_traits[c] & CharTraits.ForceStartNewTokenAfterNumber) != 0;
+                if ((s_traits[c] & CharTraits.ForceStartNewTokenAfterNumber) != 0)
+                {
+                    return true;
+                }
+
+                return forceEndNumberOnTernaryOperatorChars && (c == '?' || c == ':');
             }
```

Comment:
Is there a possible breaking change where a drive name is a number, e.x. `0x0:path`?

Reply:
Today this method is used only if the tokenizer is in `Expression` mode, so I don't see this would cause a problem to using number as drive name. But please let me know if you find a regression.
I will be sure to add tests about this.

Reply:
Note the comment about `7z` - it seems it matters for parsing command names.

Reply:
Parsing command name like `7z` or `1?2:3` works as before. `forceEndNumberOnTernaryOperatorChars` is set to true only if (1) we know for sure we are expecting an expression, and (2) that a generic token is not useful at all.